From 047bb6ca46d7cd8b2400bfbf062469ff84cec3d3 Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Fri, 20 Dec 2019 16:01:23 +0200
Subject: [PATCH] sdk_dpaa: ceetm: enable building as a module

Enable building the driver as a module by adding the required module
macros and creating the fsl_ceetm object file.

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/Makefile         | 4 +++-
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_ceetm.c | 2 ++
 2 files changed, 5 insertions(+), 1 deletion(-)

--- a/drivers/net/ethernet/freescale/sdk_dpaa/Makefile
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/Makefile
@@ -17,9 +17,11 @@ endif
 ifeq ($(CONFIG_FSL_DPAA_1588),y)
 fsl_dpa-objs += dpaa_1588.o
 endif
+
 ifeq ($(CONFIG_FSL_DPAA_CEETM),y)
 ccflags-y += -I$(srctree)/drivers/net/ethernet/freescale/sdk_fman/src/wrapper
-fsl_dpa-objs += dpaa_eth_ceetm.o
+obj-$(CONFIG_FSL_SDK_DPAA_ETH) += fsl_ceetm.o
+fsl_ceetm-objs += dpaa_eth_ceetm.o
 endif
 
 fsl_mac-objs += mac.o mac-api.o
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_ceetm.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_ceetm.c
@@ -33,6 +33,8 @@
 #include "dpaa_eth_ceetm.h"
 
 #define DPA_CEETM_DESCRIPTION "FSL DPAA CEETM qdisc"
+MODULE_LICENSE("Dual BSD/GPL");
+MODULE_DESCRIPTION(DPA_CEETM_DESCRIPTION);
 
 const struct nla_policy ceetm_policy[TCA_CEETM_MAX + 1] = {
 	[TCA_CEETM_COPT] = { .len = sizeof(struct tc_ceetm_copt) },
